src/sagemaker/jumpstart/types.py

```diff
@@ -1028,22 +1038,23 @@ class JumpStartPresetComponent(JumpStartMetadataBaseFields):
 
     __slots__ = slots + JumpStartMetadataBaseFields.__slots__
 
-    def __init__(
+    def __init__(  # pylint: disable=super-init-not-called
```

Comment:
why aren't we calling the superclass constructor? This isn't good OOP design IMO, it violates [Liskov Substution Principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle)

Reply:
I'll update, I left the component to only have its specific fields but not construct all the parent class fields which are optional, but it should work with superclass constructor. Good callout